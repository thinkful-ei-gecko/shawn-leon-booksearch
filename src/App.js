import React, { Component } from 'react';
import 'normalize.css'
import './App.css';
import Banner from './Banner/Banner';
import Results from './Results/Results';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      bookInfo : [],
    };
    }

  parseBookResults = (results) => {
    //title - results[i].volumeInfo.title
    //picture - results[i].volumeInfo.imageLinks.thumbnail
    //author - results[i].volumeInfo.authors[0]
    //price - if results[i].saleInfo.saleability !== 'FREE' 
                //then results[i].saleInfo.listPrice.amount + results[i].saleInfo.listPrice.currencyCode
              //otherwise, FREE
    //description - results[i].searchInfo.textSnippet

    let updatedState = [];
    //map out important data into the object

    results.items.forEach(book => {
      let currentBook = {};
      if (book.volumeInfo.title) { currentBook.title = book.volumeInfo.title; }
      else { currentBook.title = 'No Title' }
      if (book.volumeInfo.imageLinks) { 
        if (book.volumeInfo.imageLinks.thumbnail) { currentBook.picture = book.volumeInfo.imageLinks.thumbnail; }
        else { currentBook.picture = 'http://www.stleos.uq.edu.au/wp-content/uploads/2016/08/image-placeholder.png'; }
      }
      else { currentBook.picture = 'http://www.stleos.uq.edu.au/wp-content/uploads/2016/08/image-placeholder.png'; }
      if (book.volumeInfo.authors) { currentBook.author = book.volumeInfo.authors[0] }
      else { currentBook.author = 'No author listed' }
      if (book.saleInfo.saleability === 'FOR_SALE') {
        currentBook.price = `${book.saleInfo.listPrice.amount} ${book.saleInfo.listPrice.currencyCode}`;
      }
      else if (book.saleInfo.saleability === 'NOT_FOR_SALE' ) {
        currentBook.price = 'NOT FOR SALE';
      }
      else {
        currentBook.price = 'FREE';
      }
      if (book.searchInfo) { 
        if (book.searchInfo) { currentBook.description = book.searchInfo.textSnippet; }
        else { currentBook.description = 'no description'; }
      }
      else { currentBook.description = 'no description'; }
      updatedState.push(currentBook);
    })
    
    let newStateObject = {
      bookInfo : updatedState
    }
    console.log(newStateObject);
    this.setState({
      ...newStateObject
    })
  }
  
  APIcall = (searchTerm,printType,filter) => {
    let baseURL = `https://www.googleapis.com/books/v1/volumes?q=${searchTerm}`;
    if (filter !== 'all') { 
      baseURL += `&filter=${filter}`;
    }
    baseURL += `&printType=${printType}`;
    baseURL += `&maxResults=8`;
    console.log(baseURL);
    fetch(baseURL)
    .then(res => {
      if (res.ok) {
        return res.json();
      }
      throw new Error('could not resolve');
    })
    .then(resJson => {
      console.log(resJson);
      this.parseBookResults(resJson);
    })
    .catch(e => console.error(e));
  }


  render() {
    return (
      <div className="App">
        <Banner APIcall={this.APIcall}></Banner>
        
        <Results stateData={this.state.bookInfo}></Results>       
      </div>
    );
  }
}


export default App;
