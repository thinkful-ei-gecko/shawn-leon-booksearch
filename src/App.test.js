import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

describe('<App />', () => {

  //smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

    //snapshot with enzyme
  it('renders empty with no data', () => {
    const wrapper = shallow(<App />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

});