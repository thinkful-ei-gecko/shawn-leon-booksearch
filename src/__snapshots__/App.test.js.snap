// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders empty with no data 1`] = `
<div
  className="App"
>
  <Banner
    APIcall={[Function]}
  />
  <Results
    stateData={Array []}
  />
</div>
`;
